### PR TITLE
Captures signal from OS and stores the result in the exit code

### LINKED ISSUE #617: Investigate options to capture accurate exit-code signals

### PR DESCRIPTION
**Pull request by cppwfs**
Task now does a system exit if a sig-int is captured.

Integration test can be removed if we need to go to tests.

resolves #617

### CODE REVIEW

File: spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java [hunk 1]
```diff
@@ -444,4 +460,16 @@ public int getOrder() {
 		return HIGHEST_PRECEDENCE;
 	}
 
+	public void listenTo(String name) {
+		sun.misc.Signal signal = new sun.misc.Signal(name);
+		sun.misc.Signal.handle(signal, this);
```
[the remark this answers is not in the document]
Comment (cppwfs) [reply spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java]: You are correct.  And for imports there was no way I could see to disable that check unless we were allowed to update the XML configurations for checkstyle in the Spring Cloud project.  And at this time I don't think they will allow that check to be disabled.

File: spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java [hunk 2]
```diff
@@ -118,6 +118,12 @@
 
 	private ExitCodeEvent exitCodeEvent;
 
+	private boolean isSignalReceived;
```
[the remark this answers is not in the document]
Comment (cppwfs) [reply spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java]: fixed

File: spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java [hunk 3]
```diff
@@ -147,6 +156,10 @@ public TaskLifecycleListener(TaskRepository taskRepository,
 		this.taskProperties = taskProperties;
 		this.taskListenerExecutorObjectFactory = taskListenerExecutorObjectFactory;
 		this.taskMetrics = new TaskMetrics();
+		this.listenTo("HUP");
+		this.listenTo("INT");
+		this.listenTo("TERM");
```
[the remark this answers is not in the document]
Comment (cppwfs) [reply spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java]: done

**Comment by sabbyanandan** [comment]
We may have to revisit this functionality once when we figure out how this would work with Graal. Today, Graal doesn't work on `sun.misc` package. Closing it for now.

**Pull request closed without merging**